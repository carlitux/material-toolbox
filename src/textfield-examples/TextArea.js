// @flow
import * as React from 'react';

import FormField from '../material-toolbox/form-field';
import Example from '../Example';
import TextField from '../material-toolbox/textfield/TextField';

const source = `import TextField from 'material-toolbox/textfield/TextField';
import FormField from 'material-toolbox/form-field';

export default class TextArea extends React.Component {
  state = {
    disabled: false,
    rtl: false,
    required: false,
    alternateColor: false,
  };

  handleChange = (evt) => {
    const { name } = evt.currentTarget;
    this.setState(state => ({
      [name]: !state[name],
    }));
  };

  render() {
    return (
      <div>
        <div dir={this.state.rtl ? 'rtl' : undefined}>
          <TextField
            label="Textarea Label"
            className={
              this.state.alternateColor
                ? 'demo-text-field-custom-colors demo-text-field-custom-error-colors'
                : undefined
            }
            disabled={this.state.disabled}
            required={this.state.required}
            aria-controls="multiline-helper-text"
            rows="8"
            cols="40"
            multiline
          />
        </div>

        <div>
          <FormField label="Disabled" htmlFor="multiline-disabled">
            <input
              id="multiline-disabled"
              type="checkbox"
              name="disabled"
              checked={this.state.disabled}
              onChange={this.handleChange}
            />
          </FormField>
        </div>

        <div>
          <FormField label="RTL" htmlFor="multiline-rtl">
            <input
              id="multiline-rtl"
              type="checkbox"
              name="rtl"
              checked={this.state.rtl}
              onChange={this.handleChange}
            />
          </FormField>
        </div>

        <div>
          <FormField label="Required" htmlFor="multiline-required">
            <input
              id="multiline-required"
              type="checkbox"
              name="required"
              checked={this.state.required}
              onChange={this.handleChange}
            />
          </FormField>
        </div>

        <div>
          <FormField label="Alternate Colors" htmlFor="multiline-alternate">
            <input
              id="multiline-alternate"
              type="checkbox"
              name="alternateColor"
              checked={this.state.alternateColor}
              onChange={this.handleChange}
            />
          </FormField>
        </div>
      </div>
    );
  }
}
`;

type Props = {};
type State = {
  disabled: boolean,
  rtl: boolean,
  required: boolean,
  alternateColor: boolean,
};

export default class TextArea extends React.Component<Props, State> {
  state = {
    disabled: false,
    rtl: false,
    required: false,
    alternateColor: false,
  };

  handleChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
    const { name } = evt.currentTarget;
    this.setState(state => ({
      [name]: !state[name],
    }));
  };

  render() {
    return (
      <div>
        <Example title="Textarea" source={source}>
          <div dir={this.state.rtl ? 'rtl' : undefined}>
            <TextField
              label="Textarea Label"
              className={
                this.state.alternateColor
                  ? 'demo-text-field-custom-colors demo-text-field-custom-error-colors'
                  : undefined
              }
              disabled={this.state.disabled}
              required={this.state.required}
              aria-controls="multiline-helper-text"
              rows="8"
              cols="40"
              multiline
            />
          </div>

          <div>
            <FormField label="Disabled" htmlFor="multiline-disabled">
              <input
                id="multiline-disabled"
                type="checkbox"
                name="disabled"
                checked={this.state.disabled}
                onChange={this.handleChange}
              />
            </FormField>
          </div>

          <div>
            <FormField label="RTL" htmlFor="multiline-rtl">
              <input
                id="multiline-rtl"
                type="checkbox"
                name="rtl"
                checked={this.state.rtl}
                onChange={this.handleChange}
              />
            </FormField>
          </div>

          <div>
            <FormField label="Required" htmlFor="multiline-required">
              <input
                id="multiline-required"
                type="checkbox"
                name="required"
                checked={this.state.required}
                onChange={this.handleChange}
              />
            </FormField>
          </div>

          <div>
            <FormField label="Alternate Colors" htmlFor="multiline-alternate">
              <input
                id="multiline-alternate"
                type="checkbox"
                name="alternateColor"
                checked={this.state.alternateColor}
                onChange={this.handleChange}
              />
            </FormField>
          </div>
        </Example>
      </div>
    );
  }
}
