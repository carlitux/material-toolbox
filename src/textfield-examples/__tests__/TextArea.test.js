// @flow
import React from 'react';
import { shallow } from 'enzyme';
import TextArea from '../TextArea';

describe('example::TextArea', () => {
  it('Should match snapshoot', () => {
    const component = shallow(<TextArea />);
    expect(component).toMatchSnapshot();
  });
});
