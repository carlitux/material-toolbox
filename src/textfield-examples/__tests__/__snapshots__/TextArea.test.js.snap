// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example::TextArea Should match snapshoot 1`] = `
<div>
  <Example
    source="import TextField from 'material-toolbox/textfield/TextField';
import FormField from 'material-toolbox/form-field';

export default class TextArea extends React.Component {
  state = {
    disabled: false,
    rtl: false,
    required: false,
    alternateColor: false,
  };

  handleChange = (evt) => {
    const { name } = evt.currentTarget;
    this.setState(state => ({
      [name]: !state[name],
    }));
  };

  render() {
    return (
      <div>
        <div dir={this.state.rtl ? 'rtl' : undefined}>
          <TextField
            label=\\"Textarea Label\\"
            className={
              this.state.alternateColor
                ? 'demo-text-field-custom-colors demo-text-field-custom-error-colors'
                : undefined
            }
            disabled={this.state.disabled}
            required={this.state.required}
            aria-controls=\\"multiline-helper-text\\"
            rows=\\"8\\"
            cols=\\"40\\"
            multiline
          />
        </div>

        <div>
          <FormField label=\\"Disabled\\" htmlFor=\\"multiline-disabled\\">
            <input
              id=\\"multiline-disabled\\"
              type=\\"checkbox\\"
              name=\\"disabled\\"
              checked={this.state.disabled}
              onChange={this.handleChange}
            />
          </FormField>
        </div>

        <div>
          <FormField label=\\"RTL\\" htmlFor=\\"multiline-rtl\\">
            <input
              id=\\"multiline-rtl\\"
              type=\\"checkbox\\"
              name=\\"rtl\\"
              checked={this.state.rtl}
              onChange={this.handleChange}
            />
          </FormField>
        </div>

        <div>
          <FormField label=\\"Required\\" htmlFor=\\"multiline-required\\">
            <input
              id=\\"multiline-required\\"
              type=\\"checkbox\\"
              name=\\"required\\"
              checked={this.state.required}
              onChange={this.handleChange}
            />
          </FormField>
        </div>

        <div>
          <FormField label=\\"Alternate Colors\\" htmlFor=\\"multiline-alternate\\">
            <input
              id=\\"multiline-alternate\\"
              type=\\"checkbox\\"
              name=\\"alternateColor\\"
              checked={this.state.alternateColor}
              onChange={this.handleChange}
            />
          </FormField>
        </div>
      </div>
    );
  }
}
"
    title="Textarea"
  >
    <div>
      <TextField
        aria-controls="multiline-helper-text"
        box={false}
        cols="40"
        cssOnly={false}
        dense={false}
        disabled={false}
        fullWidth={false}
        invalid={false}
        label="Textarea Label"
        multiline={true}
        outlined={false}
        persistentHelperText={false}
        required={false}
        rows="8"
        validationHelperText={false}
      />
    </div>
    <div>
      <FormField
        alignEnd={false}
        cssOnly={false}
        htmlFor="multiline-disabled"
        label="Disabled"
      >
        <input
          checked={false}
          id="multiline-disabled"
          name="disabled"
          onChange={[Function]}
          type="checkbox"
        />
      </FormField>
    </div>
    <div>
      <FormField
        alignEnd={false}
        cssOnly={false}
        htmlFor="multiline-rtl"
        label="RTL"
      >
        <input
          checked={false}
          id="multiline-rtl"
          name="rtl"
          onChange={[Function]}
          type="checkbox"
        />
      </FormField>
    </div>
    <div>
      <FormField
        alignEnd={false}
        cssOnly={false}
        htmlFor="multiline-required"
        label="Required"
      >
        <input
          checked={false}
          id="multiline-required"
          name="required"
          onChange={[Function]}
          type="checkbox"
        />
      </FormField>
    </div>
    <div>
      <FormField
        alignEnd={false}
        cssOnly={false}
        htmlFor="multiline-alternate"
        label="Alternate Colors"
      >
        <input
          checked={false}
          id="multiline-alternate"
          name="alternateColor"
          onChange={[Function]}
          type="checkbox"
        />
      </FormField>
    </div>
  </Example>
</div>
`;
